instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/GrpcTest.java

```diff
@@ -56,6 +59,83 @@ protected InstrumentationExtension testing() {
     return testing;
   }
 
+  @Test
+  void grpcAttributesExtractor() throws Exception {
+    String metadataKey = "some-key";
+    AttributeKey<List<String>> attributeKey =
+        AttributeKey.stringArrayKey(METADATA_ATTRIBUTE_PREFIX + metadataKey);
+    String metadataValue = "some-value";
+    List<String> metadataValueAsList = Collections.singletonList("some-value");
+
+    BindableService greeter =
+        new GreeterGrpc.GreeterImplBase() {
+          @Override
+          public void sayHello(
+              Helloworld.Request req, StreamObserver<Helloworld.Response> responseObserver) {
+            Helloworld.Response reply =
+                Helloworld.Response.newBuilder().setMessage("Hello " + req.getName()).build();
+            responseObserver.onNext(reply);
+            responseObserver.onCompleted();
+          }
+        };
+
+    GrpcAttributesExtractor grpcAttributesExtractor =
+        new GrpcAttributesExtractor(
+            GrpcRpcAttributesGetter.INSTANCE, Collections.singletonList(metadataKey));
+
+    Server server =
+        ServerBuilder.forPort(0)
+            .addService(greeter)
+            .intercept(
+                GrpcTelemetry.builder(testing.getOpenTelemetry())
+                    .addAttributeExtractor(grpcAttributesExtractor)
+                    .build()
+                    .newServerInterceptor())
+            .build()
+            .start();
+
+    ManagedChannel channel =
+        createChannel(
+            ManagedChannelBuilder.forAddress("localhost", server.getPort())
+                .intercept(
+                    GrpcTelemetry.builder(testing.getOpenTelemetry())
+                        .addAttributeExtractor(grpcAttributesExtractor)
```

Comment:
Shouldn't you test the new `setCapturedRequestMetadata` method instead?

Reply:
done, i'm intercepting it at the GrpcTest